extensions/hotreload/hotreload.go

```diff
@@ -0,0 +1,104 @@
+package hotreload
+
+import (
+    "fmt"
+    "html/template"
+    "log"
+    "sync"
+
+    _ "embed"
+
+    . "github.com/emad-elsaid/xlog"
+    "github.com/gorilla/websocket"
+)
+
+var (
+    upgrader     = websocket.Upgrader{ReadBufferSize: 1024, WriteBufferSize: 1024}
+    clients      = make(map[*websocket.Conn]bool)
+    clientsMutex sync.Mutex
+)
+
+func init() {
+    if !READONLY {
+        Listen(Changed, NotifyPageChange)
+        Get(`/ws`, handleWebSocket)
+        RegisterWidget(AFTER_VIEW_WIDGET, 0, clientWidget)
+    }
+}
+
+
+func NotifyPageChange(p Page) error {
+    if !p.Exists() {
+        return nil
+    }
+
+    message := map[string]string{"url": fmt.Sprintf("/%s", p.Name())}
+    log.Printf("Page %s changed from notifiy", p.Name())
+
+    clientsMutex.Lock()
+    defer clientsMutex.Unlock()
+
+    for client := range clients {
+        err := client.WriteJSON(message)
+        if err != nil {
+            log.Printf("Error sending message to client: %v\n", err)
+            client.Close()
+            delete(clients, client)
+        }
+    }
+    return nil
+}
+
+func nop(w Response, r Request) {
```

Comment:
:eyes: I'm not sure if this is required ? Do you want maybe `xlog.NoContent` ?

Reply:
`xlog.NoContent` injects a http header to the response, then not being able to convert the request to a websocket. so I just made a new real nop

Reply:
I see, Can we add this to `server.go` and export it as `Noop` so it can be used in other cases?